### PR TITLE
8.x field config

### PR DESCRIPTION
**Pull request by qadan**
Storage of field configuration has been moved to config instead of the DB.

An update hook provides some coverage for moving this over.

Some functionality has been set in place to ensure ease of translatability

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,12 @@ This module requires the following modules/libraries:
 
 ## Installation
 
+Existing Solr fields configuration from Drupal 7 can be migrated by dropping the
```
Comment (willtp87) [inline README.md]: This documentation is probably not needed. If someone is following the standard Drupal upgrade paths it should just work and if they are off the beaten path then the docs are out of scope.

File: includes/admin.inc [hunk 2]
```diff
@@ -95,6 +96,13 @@ function islandora_solr_admin_settings_fields(FormStateInterface $form_state, ar
       foreach ($form_state_terms as $key => $value) {
         if ($field == $value['solr_field']) {
           unset($form_state_terms[$key]);
+          $field_key = ConfigFieldFormBase::generateFieldKey($value['solr_field']);
+          $config = \Drupal::configFactory()
+            ->getEditable('islandora_solr.fields');
+          $fields = $config->get($field_type);
+          unset($fields[$field_key]);
+          $config->set($field_type, $fields)
+            ->save();
```
Comment (willtp87) [inline includes/admin.inc]: Isn't modifying persistent data here a departure from existing behavior?

File: src/Form/IslandoraDateFilter.php [hunk 3]
```diff
@@ -166,7 +166,7 @@ public function buildForm(array $form, FormStateInterface $form_state, array $el
       '#name' => 'date_filter_facet_field_' . $form_key,
       '#form_key' => $form_key,
     ];
-    $form['#attached']['library'][] = 'core/jquery.ui.datepicker';
+    $form['#attached']['library'][] = 'islandora_solr/islandora_solr_facets.datepicker.js';
```
Comment (willtp87) [inline src/Form/IslandoraDateFilter.php]: This is not a lib? `islandora_solr/datepicker` maybe? If this is so, it questions how the related work was tested.

Comment (qadan) [reply src/Form/IslandoraDateFilter.php]: I'm dumb, I was looking for the date slider and not the datepicker during testing of console logs

**Pull request merged**